File: 070. Climbing Stairs.swift
class Solution {
    func climbStairs(_ n: Int) -> Int {
        var map = Dictionary<Int, Int>()

        map[0] = 1
        map[1] = 2
        
        if n < 2 {
            return map[n-1]!
        }
        
        for i in 2 ..< n {
            map[i] = map[i - 1]! + map[i - 2]!
        }
        
        return map[n-1]!
    }
}

File: 070. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        
        mapping = {}
        mapping[0] = 0
        mapping[1] = 1
        mapping[2] = 2
        
        for i in range(3, n+1):
            mapping[i] = mapping[i-1] + mapping[i-2]
        
        return mapping[n]
